package-system/openimageio-opencolorio/test/test_OpenImageIO.cpp

```diff
@@ -60,11 +61,97 @@ int main()
     if (!inp->read_image(0,0, 0, 0, TypeDesc::UINT8, &pixels[0]))
     {
         printf("Unable to read pixels from image.\n");
-        return -1;    
-    
+        return false;
     }
 
     inp->close();
+    return true;
+}
+
+bool testWritingImage()
+{
+    using namespace OIIO;
+     OIIO::TypeDesc pixelFormat = OIIO::TypeDesc::UINT8;
+    std::unique_ptr<OIIO::ImageOutput> outputImage = OIIO::ImageOutput::create("temp_save_image.png");
+    if (!outputImage)
+    {
+        printf("Failed to create the OIIO::ImageOutput.\n");
+        return false;
+    }
+
+    OIIO::ImageSpec spec(1024, 1024, 4, pixelFormat);
+    if (!outputImage->open("temp_save_image.png", spec))
+    {
+        printf("Failed to open temp_save_image.png\n");
+        return false;
+    }
+
+    std::vector<unsigned char> pixelBuffer;
+    pixelBuffer.resize(1024 * 1024 * 4);
+    memset(pixelBuffer.data(), 0, 1024 * 1024 * 4);
+    if (!outputImage->write_image(pixelFormat, pixelBuffer.data()))
+    {
+        printf("Failed to write temp_save_image.png\n");
+        return false;
+    }
+
+    outputImage->close();
+
+    return true;
+}
+
+
+bool testWritingImage_tif_float() // one channel 32-bit-float (4 bytes per channel)
+{
+    using namespace OIIO;
+     OIIO::TypeDesc pixelFormat = OIIO::TypeDesc::FLOAT;
+    std::unique_ptr<OIIO::ImageOutput> outputImage = OIIO::ImageOutput::create("temp_save_image.tif");
+    if (!outputImage)
+    {
+        printf("Failed to create the OIIO::ImageOutput.\n");
+        return false;
+    }
+
+    OIIO::ImageSpec spec(1024, 1024, 1, pixelFormat);
+    if (!outputImage->open("temp_save_image.tif", spec))
+    {
+        printf("Failed to open temp_save_image.tif\n");
+        return false;
+    }
+
+    std::vector<unsigned char> pixelBuffer;
+    pixelBuffer.resize(1024 * 1024 * 4);
+    memset(pixelBuffer.data(), 0, 1024 * 1024 * 4);
+    if (!outputImage->write_image(pixelFormat, pixelBuffer.data()))
+    {
+        printf("Failed to write temp_save_image.tif\n");
+        return false;
+    }
+
+    outputImage->close();
+
+    return true;
+}
+
+
+int main()
+{
+    if (!testReadingImage())
+    {
+        return 1;
+    }
+
+    if (!testWritingImage())
+    {
+        return 1;
+    }
+
+    if (!testWritingImage_tif_float())
+    {
+        return 1;
+    }
+
```

Comment:
Shouldn't the tests clean up any temp images they write?

Reply:
Eh, yeah, fair point. But at least making it clean up in the common case would be nice. :)